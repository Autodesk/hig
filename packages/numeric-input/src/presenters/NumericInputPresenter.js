import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";

import { ThemeContext } from "@hig/theme-context";

import stylesheet from "./NumericInputPresenter.stylesheet";
import { availableVariants } from "../constants";

function NumericInputPresenter(props) {
  const {
    children,
    hasFocus,
    hasHover,
    isDisabled,
    stylesheet: customStylesheet,
    variant
  } = props;

  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles, metadata }) => {
        const styles = stylesheet(
          { isDisabled, hasFocus, hasHover, variant },
          resolvedRoles
        );
        const cssStyles = customStylesheet
          ? customStylesheet(
              styles,
              props,
              resolvedRoles,
              metadata.colorSchemeId
            )
          : styles;

        return <div className={css(cssStyles.wrapper)}>{children}</div>;
      }}
    </ThemeContext.Consumer>
  );
}

NumericInputPresenter.propTypes = {
  children: PropTypes.node,
  hasFocus: PropTypes.bool,
  hasHover: PropTypes.bool,
  isDisabled: PropTypes.bool,
  stylesheet: PropTypes.func,
  variant: PropTypes.oneOf(availableVariants)
};

export default NumericInputPresenter;
