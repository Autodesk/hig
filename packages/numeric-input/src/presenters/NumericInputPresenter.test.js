import React from "react";
import renderer from "react-test-renderer";

import NumericInputPresenter from "./NumericInputPresenter";

describe("NumericInputPresenter", () => {
  it("renders by default", () => {
    expect(
      renderer.create(<NumericInputPresenter />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders with id", () => {
    expect(
      renderer.create(<NumericInputPresenter id="my_id" />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders with hover", () => {
    expect(
      renderer.create(<NumericInputPresenter hasHover />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders with focus", () => {
    expect(
      renderer.create(<NumericInputPresenter hasFocus />).toJSON()
    ).toMatchSnapshot();
  });

  it("renders when disabled", () => {
    expect(
      renderer.create(<NumericInputPresenter isDisabled />).toJSON()
    ).toMatchSnapshot();
  });
});
