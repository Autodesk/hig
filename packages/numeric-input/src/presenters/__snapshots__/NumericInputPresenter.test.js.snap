// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInputPresenter renders by default 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="emotion-0"
/>
`;

exports[`NumericInputPresenter renders when disabled 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="emotion-0"
/>
`;

exports[`NumericInputPresenter renders with focus 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="emotion-0"
/>
`;

exports[`NumericInputPresenter renders with hover 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="emotion-0"
/>
`;

exports[`NumericInputPresenter renders with id 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="emotion-0"
/>
`;
